digitalocean/database/resource_database_mongodb_config.go

```diff
@@ -0,0 +1,174 @@
+package database
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/digitalocean/godo"
+	"github.com/digitalocean/terraform-provider-digitalocean/digitalocean/config"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceDigitalOceanDatabaseMongoDBConfig() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDigitalOceanDatabaseMongoDBConfigCreate,
+		ReadContext:   resourceDigitalOceanDatabaseMongoDBConfigRead,
+		UpdateContext: resourceDigitalOceanDatabaseMongoDBConfigUpdate,
+		DeleteContext: resourceDigitalOceanDatabaseMongoDBConfigDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceDigitalOceanDatabaseMongoDBConfigImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"cluster_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"connect_default_read_concern": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice(
+					[]string{
+						"local",
+						"available",
+						"majority",
+					},
+					false,
+				),
```

Comment:
The API seems to always return this field in uppercase which can cause unexpected diffs like:

```
Terraform will perform the following actions:

  # digitalocean_database_mongodb_config.example will be updated in-place
  ~ resource "digitalocean_database_mongodb_config" "example" {
      ~ default_read_concern               = "MAJORITY" -> "majority"
        id                                 = "a12903dc-13bf-4ee3-8995-3d0507a4310e/mongodb-config"
        # (5 unchanged attributes hidden)
    }
```

We can workaround this by seeing `ignoreCase` to `true` in the `ValidateFunc` and by using an `DiffSuppressFunc` to ignore the case when comparing the local and remote values.

https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-behaviors#diffsuppressfunc

```suggestion
				ValidateFunc: validation.StringInSlice(
					[]string{
						"local",
						"available",
						"majority",
					},
					true,
				),
				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
					return strings.EqualFold(old, new)
				},
```

Reply:
`DiffSuppressFunc` definitely helps :+1: 
Thanks